packages/report-server/src/reportBuilder/transform/functions/mergeRows/mergeRows.ts

```diff
@@ -70,19 +70,22 @@ const groupRows = (rows: Row[], params: MergeRowsParams) => {
       return;
     }
     const groupKey = params.createGroupKey(row);
-    groupsByKey[groupKey] = addRowToGroup(groupsByKey[groupKey] || {}, row);
+    groupsByKey[groupKey] = addRowToGroup(groupsByKey[groupKey], row);
```

Comment:
I reckon I'd prefer a function signature that emphasised that it was mutating internally, because the assignment here makes you think it wouldn't be
```suggestion
    addRowToGroup(groupsByKey, groupKey, row); // mutates groupsByKey
```

Reply:
Ah good call, will do!

Reply:
Great optimisation btw, thanks for improving this!

Reply:
I was testing with a report that had 70k rows, and the performance was really bad (+1 min to return). Initially I went pale thinking that mathjs expression parsing was too slow... moral of the story is to check yourself before you blame the 3rd party lib 😂